### PR TITLE
Fixed Firestone Ore XP and Fortune duplication

### LINKED ISSUE #1114: Infinite XP from Firestone Ore
Mining the firestone ore gives you the ore and experience, you can just keep placing it down to get more experience. I know the whole firestone thing isn't really done yet, but I would be great if it wouldn't drop experience.

Railcraft_1.10.2-10.0.1
Modpack: FTB Beyond 1.1.0

### PR DESCRIPTION
**Pull request by Generalcamo**
Closes #1114

### CODE REVIEW

File: src/main/java/mods/railcraft/common/blocks/ore/BlockOreMagic.java [hunk 1]
```diff
@@ -89,6 +77,47 @@ public void randomDisplayTick(IBlockState stateIn, World worldIn, BlockPos pos,
...
             EffectManager.instance.fireSparkEffect(worldIn, startPosition, endPosition);
+            this.spawnParticles(worldIn, pos);
+        }
+    }
+
+    private void spawnParticles(World worldIn, BlockPos pos) {
+        Random random = worldIn.rand;
+        double d0 = 0.0625D;
+
+        for (int i = 0; i < 6; ++i) {
```
Comment (liach) [inline src/main/java/mods/railcraft/common/blocks/ore/BlockOreMagic.java]:
```java
for (EnumFacing face : EnumFacing.values()) {
```

File: src/main/java/mods/railcraft/common/blocks/ore/BlockOreMagic.java [hunk 2]
```diff
@@ -89,6 +77,47 @@ public void randomDisplayTick(IBlockState stateIn, World worldIn, BlockPos pos,
...
+
+            if (i == 4 && !worldIn.getBlockState(pos.east()).isOpaqueCube()) {
+                d1 = (double) pos.getX() + 0.0625D + 1.0D;
+            }
+
+            if (i == 5 && !worldIn.getBlockState(pos.west()).isOpaqueCube()) {
+                d1 = (double) pos.getX() - 0.0625D;
+            }
+
+            if (d1 < (double) pos.getX() || d1 > (double) (pos.getX() + 1) || d2 < 0.0D || d2 > (double) (pos.getY() + 1) || d3 < (double) pos.getZ() || d3 > (double) (pos.getZ() + 1)) {
```
Comment (liach) [inline src/main/java/mods/railcraft/common/blocks/ore/BlockOreMagic.java:117]: Why spawn particle only when d is not within the block position?

File: src/main/java/mods/railcraft/common/blocks/ore/BlockOreMagic.java [hunk 3]
```diff
@@ -89,6 +77,47 @@ public void randomDisplayTick(IBlockState stateIn, World worldIn, BlockPos pos,
...
+            this.spawnParticles(worldIn, pos);
+        }
+    }
+
+    private void spawnParticles(World worldIn, BlockPos pos) {
+        Random random = worldIn.rand;
+        double d0 = 0.0625D;
+
+        for (int i = 0; i < 6; ++i) {
+            double d1 = (double) ((float) pos.getX() + random.nextFloat());
```
Comment (liach) [inline src/main/java/mods/railcraft/common/blocks/ore/BlockOreMagic.java:89]: Just `pos.getX() + random.nextDouble()`

File: src/main/java/mods/railcraft/common/blocks/ore/BlockOreMagic.java [hunk 4]
```diff
@@ -89,6 +77,47 @@ public void randomDisplayTick(IBlockState stateIn, World worldIn, BlockPos pos,
...
+
+            if (i == 3 && !worldIn.getBlockState(pos.north()).isOpaqueCube()) {
+                d3 = (double) pos.getZ() - 0.0625D;
+            }
+
+            if (i == 4 && !worldIn.getBlockState(pos.east()).isOpaqueCube()) {
+                d1 = (double) pos.getX() + 0.0625D + 1.0D;
+            }
+
+            if (i == 5 && !worldIn.getBlockState(pos.west()).isOpaqueCube()) {
```
Comment (liach) [inline src/main/java/mods/railcraft/common/blocks/ore/BlockOreMagic.java:113]: These checks look really ugly.

Comment (Generalcamo) [reply src/main/java/mods/railcraft/common/blocks/ore/BlockOreMagic.java:113]: It's the exact checks used by Vanilla Redstone Ore.

Comment (liach) [reply src/main/java/mods/railcraft/common/blocks/ore/BlockOreMagic.java:113]: Redstone ore only do this when they are walked on or clicked; should you add a blockstate and do this?

**Review by liach — APPROVED** [review]

**Comment by liach** [comment]
Since you said you copied this code from vanilla redstone, I will leave this to CovertJaguar. The build failed because maven was down, so I will restart it.

**Review by CovertJaguar — APPROVED** [review]
Looks good, though I might clean up the particle code a bit.

I see what I did wrong when I refactored.

**Pull request merged**